src/backend/joanie/core/api/client.py

```diff
@@ -51,7 +51,7 @@ def get_queryset(self):
             raise NotFound("The requested resource was not found on this server.")
 
         if course_id:
-            queryset.filter(course_id=self.kwargs["course_id"])
+            queryset = queryset.filter(course=course_id)
```

Comment:
I did see the need of the select_related in the main queryset:
```py
queryset = models.CourseRun.objects.filter(is_listed=True).select_related("course")
```
CourseRunSerializer doesn't use course fields, we could remove this join.

Also, when i try to fix this i've added this: 
```py
if not course_id:
    course_id = self.request.query_params.get("course_id")
if self.action == "list" and not course_id:
    raise NotFound("The requested resource was not found on this server.")
 ```
 This way the viewset can be used with `course-runs/?course_id={course_id}`